comp/core/remoteagentregistry/impl/remoteagentregistry.go

```diff
@@ -178,6 +187,77 @@ func (ra *remoteAgentRegistry) start() {
 	}()
 }
 
+type registryCollector struct {
+	agentMapMu   *sync.Mutex
+	agentMap     map[string]*remoteAgentDetails
+	queryTimeout time.Duration
+}
+
+func NewRegistryCollector(agentMapMu *sync.Mutex, agentMap map[string]*remoteAgentDetails, queryTimeout time.Duration) prometheus.Collector {
+	return &registryCollector{
+		agentMapMu:   agentMapMu,
+		agentMap:     agentMap,
+		queryTimeout: queryTimeout,
+	}
+}
+
+func (c *registryCollector) Describe(ch chan<- *prometheus.Desc) {
+}
+
+func (c *registryCollector) Collect(ch chan<- prometheus.Metric) {
+	c.getRegisteredAgentsTelemetry(ch)
+}
+
+func (c *registryCollector) getRegisteredAgentsTelemetry(_ chan<- prometheus.Metric) {
+	c.agentMapMu.Lock()
+
+	agentsLen := len(c.agentMap)
+
+	// Return early if we have no registered remote agents.
+	if agentsLen == 0 {
+		c.agentMapMu.Unlock()
+		return
+	}
+
+	data := make(chan *pb.GetTelemetryResponse, agentsLen)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for agentID, details := range c.agentMap {
+		go func() {
+			// Retrieve the telemetry data in exposition format from the remote agent
+			resp, err := details.client.GetTelemetry(ctx, &pb.GetTelemetryRequest{}, grpc.WaitForReady(true))
+			if err != nil {
+				log.Warnf("Failed to query remote agent '%s' for telemetry data: %v", agentID, err)
+				return
+			}
+			// TODO: parse the response and add the metrics to the channel
```

Comment:
The idea is to retrieve the text response from the grpc call and parsing it with prometheus `expfmt`.